.github/workflows/automated-updates-to-sam-cli.yml

```diff
@@ -0,0 +1,104 @@
+          path: aws-sam-cli
+
+      - name: Update hash & commit
+        run: |
+          git config --global user.email "action@github.com"
+          git config --global user.name "GitHub Action"
+          cd aws-sam-cli-app-templates
+          APP_TEMPLATES_COMMIT_HASH=$(git rev-parse HEAD)
+          cd ../aws-sam-cli
+          git checkout -b update_app_templates_hash
+          git reset --hard develop
+          cat <<< "$(jq --arg commit_hash "$APP_TEMPLATES_COMMIT_HASH" --indent 4 '.app_template_repo_commit =  $commit_hash' samcli/runtime_config.json)" > samcli/runtime_config.json
+          git status
+          git diff --quiet && exit 0 # exit if there is no change
+          echo "is_hash_changed=1" >> $GITHUB_ENV # set env variable for next step run decision
+          git add -u
+          git commit -m "feat: updating app templates repo hash with ($APP_TEMPLATES_COMMIT_HASH)"
+
+      - name: Raise PR for SAM CLI
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        if: ${{ env.is_hash_changed == 1 }} # run only if there was a change
+        run: |
+          cd aws-sam-cli
+          git push --force origin update_app_templates_hash
+          gh pr list --repo aws/aws-sam-cli --head update_app_templates_hash --json id --jq length | grep 1 && exit 0 # exit if there is existing pr
+          gh pr create --base develop --head update_app_templates_hash --title "feat: update SAM CLI with latest App Templates commit hash" --body "This PR & commit is automatically created from App Templates repo to update the SAM CLI with latest hash of the App Templates." --label "pr/internal"
+
+  updateSAMTranslator:
+    permissions:
+      pull-requests: write
+      contents: write
+    if: github.repository == 'aws/aws-sam-cli'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout SAM
+        uses: actions/checkout@v3
+        with:
+          repository: aws/serverless-applicaiton-model
+          path: serverless-applicaiton-model
+          ref: main
+      
+      - name: Checkout SAM CLI
+        uses: actions/checkout@v3
+        with:
+          repository: aws/aws-sam-cli
+          path: aws-sam-cli
+      
+      - uses: actions/setup-python@v4 # used for make update-reproducible-reqs below
+        with:
+          python-version: "3.7"
+
+      - name: Update aws-sam-translator & commit
+        run: |
+          git config --global user.email "action@github.com"
+          git config --global user.name "GitHub Action"
+          cd serverless-applicaiton-model
+          SAM_T_CUR_VERSION=$(git describe --tags --abbrev=0 | sed 's/v//')
+          cd ../aws-sam-cli
+          git checkout -b update_sam_transform_version
+          SAM_T_PRE_VERSION=$(grep "aws-sam-translator=" requirements/base.txt)
+          git reset --hard develop
+          sed -i "s/aws-sam-translator=$SAM_T_CUR_VERSION/$SAM_T_PRE_VERSION/g" requirements/base.txt; make update-reproducible-reqs
```

Comment:
how can we make sure that SAM-T is already released, and it is not only an update in the SAM-T repo, and the team is still working on the release process.

Reply:
This job creates a PR. The PR tests will not pass until SAM-T is fully released to PyPi. So I think this is sufficient as is but let me know if you disagree further and we can discuss more.